.github/workflows/unittests.yml

```diff
@@ -0,0 +1,35 @@
+name: Python Unit Tests
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    name: Python unit tests
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - name: Install Poetry
+      run: |
+        curl -sSL https://install.python-poetry.org | python3 -
+    - name: Install dependencies
+      run: |
+        poetry install
+    - name: Test with pytest
+      env:
+          OPENWEATHER_API_KEY: ${{ secrets.OPENWEATHER_API_KEY }}
+          OPENWEATHER_API_URL: ${{ secrets.OPENWEATHER_API_URL }}
```

Comment:
This shouldn't be working right now since I don't have permission to add these

Reply:
what the shit, it's working

Reply:
oh yeah, cuz we're not making any real requests since it's unit tests and not integration tests... yet